### PR TITLE
Implement horse inventory

### LINKED ISSUE #4029: Can't open horse inventory
Client version:   cant find, downloaded september 12
Server OS:        Linux
Commit id:        Server running via screen disconnect, cant view console

### Expected behavior
while riding horse, E can be pressed to open invnetory, and shows horse inventory

### Actual behavior
does nothing

### Steps to reproduce the behavior
Tame horse, saddle horse, mount horse, press E

### PR DESCRIPTION
**Pull request by peterbell10**
Fixes #4029.

I've tested with minecraft 1.8 and 1.12.2.

### CODE REVIEW

File: src/Protocol/Protocol_1_8.cpp [hunk 1]
```diff
@@ -1678,7 +1679,8 @@ void cProtocol_1_8_0::SendWindowOpen(const cWindow & a_Window)
 
 	if (a_Window.GetWindowType() == cWindow::wtAnimalChest)
 	{
-		Pkt.WriteBEInt32(0);  // TODO: The animal's EntityID
+		UInt32 HorseID = static_cast<const cHorseWindow&>(a_Window).GetHorseID();
```
Comment (madmaxoft) [inline src/Protocol/Protocol_1_8.cpp]: Space before `&`

File: src/Protocol/Protocol_1_9.cpp [hunk 2]
```diff
@@ -1723,7 +1724,8 @@ void cProtocol_1_9_0::SendWindowOpen(const cWindow & a_Window)
 
 	if (a_Window.GetWindowType() == cWindow::wtAnimalChest)
 	{
-		Pkt.WriteBEInt32(0);  // TODO: The animal's EntityID
+		UInt32 HorseID = static_cast<const cHorseWindow&>(a_Window).GetHorseID();
```
Comment (madmaxoft) [inline src/Protocol/Protocol_1_9.cpp]: Space

**Comment by madmaxoft** [comment]
Does this mean that the horse UI can only be opened while riding the horse? I though you can open it at any time, jsut standing beside the horse. Obviously, I haven't really played Minecraft for ages :)

**Comment by peterbell10** [comment]
It can be opened without riding.  You just need to crouch and right click a tamed horse.
[Relevant code.](https://github.com/peterbell10/cuberite/blob/08ccdd97df69b03691b04a1138f20179551cd5d5/src/Mobs/Horse.cpp#L124-L128)

**Pull request merged**